mitiq/zne/scaling/folding.py

```diff
@@ -421,6 +289,262 @@ def fold_all(
     return folded
 
 
+# Global folding function
+@noise_scaling_converter
+def fold_global(
+    circuit: QPROGRAM, scale_factor: float, **kwargs: Any
+) -> QPROGRAM:
+    """Returns a new circuit obtained by folding the global unitary of the
+    input circuit.
+
+    The returned folded circuit has a number of gates approximately equal to
+    scale_factor * len(circuit).
+
+    Args:
+        circuit: Circuit to fold.
+        scale_factor: Factor to scale the circuit by.
+
+    Keyword Args:
+        squash_moments (bool): If True, all gates (including folded gates) are
+            placed as early as possible in the circuit. If False, new moments
+            are created for folded gates. This option only applies to QPROGRAM
+            types which have a "moment" or "time" structure. Default is True.
+        return_mitiq (bool): If True, returns a Mitiq circuit instead of
+            the input circuit type (if different). Default is False.
+
+    Returns:
+        folded: the folded quantum circuit as a QPROGRAM.
+    """
+    _check_foldable(circuit)
+
+    if not (scale_factor >= 1):
+        raise ValueError("The scale factor must be a real number >= 1.")
+
+    folded = deepcopy(circuit)
+    measurements = _pop_measurements(folded)
+    base_circuit = deepcopy(folded)
+
+    # Determine the number of global folds and the final fractional scale
+    num_global_folds, fraction_scale = divmod(scale_factor - 1, 2)
+    # Do the global folds
+    for _ in range(int(num_global_folds)):
+        folded += Circuit(inverse(base_circuit), base_circuit)
+
+    # Fold remaining gates until the scale is reached
+    operations = list(base_circuit.all_operations())
+    num_to_fold = int(round(fraction_scale * len(operations) / 2))
+
+    if num_to_fold > 0:
+        folded += Circuit(
+            [inverse(operations[-num_to_fold:])], [operations[-num_to_fold:]]
+        )
+
+    _append_measurements(folded, measurements)
+    if not (kwargs.get("squash_moments") is False):
+        folded = _squash_moments(folded)
+    return folded
+
+
+def _create_weight_mask(
+    circuit: Circuit, fidelities: Optional[Dict[str, float]],
+) -> List[float]:
+    """Returns a list of weights associated to each gate if the input
+    circuit. Measurement gates are ignored.
+
+    Args:
+        circuit: The circuit from which a weight mask is created.
+        fidelities: The dictionary of gate fidelities.
+            If None, default fidelities will be used. See the
+            docstring of local folding function for mode details.
+
+    Returns: The list of weights associated to all the gates.
+    """
+    if fidelities and not all(0.0 < f <= 1.0 for f in fidelities.values()):
+        raise ValueError("Fidelities should be in the interval (0, 1].")
+    weights = None
+    if fidelities:
+        # Round to avoid ugly numbers like 0.09999999999999998 instead of 0.1
+        # https://docs.python.org/2/tutorial/floatingpoint.html
```

Comment:
Since we're python 3+ I would remove or update this link. Seems more or less self-explanatory without a reference though.